### PR TITLE
(maint) Add 'temp-file-name' function

### PR DESCRIPTION
**Pull request by KevinCorcoran**

### CODE REVIEW

File: src/puppetlabs/kitchensink/core.clj [hunk 1]
```diff
@@ -602,6 +603,11 @@ to be a zipper."
 
 ;; ## Temp files
 
+(defn temp-file-name
+  "Returns a unique name to a temporary file, but does not actually create the file."
+  [file-name-prefix]
+  (io/file (fs/tmpdir) (fs/temp-name file-name-prefix)))
```
Comment (rlinehan) [inline src/puppetlabs/kitchensink/core.clj]: Why use `io/file` rather than `fs/file`?

Comment (KevinCorcoran) [inline src/puppetlabs/kitchensink/core.clj]: Can't sneak anything by you, can I?  :smiling_imp: 

`fs/file` does more than is necessary here, including possibly returning a file underneath cwd.  It eventually calls `io/file` regardless, so this is just a simplification.

Comment (rlinehan) [inline src/puppetlabs/kitchensink/core.clj]: Cool, thanks for the explanation!

**Comment by rlinehan** [comment]
I'm :+1: on this. I think it's a useful addition to kitchensink.

**Comment by fpringvaldsen** [comment]
:+1:

**Pull request merged**